xbstrap/base.py

```diff
@@ -2086,6 +2086,18 @@ def checkout_src(cfg, src, settings):
             else:
                 if fixed_commit is not None:
                     commit = fixed_commit
+            missing_commit = False
+            try:
+                subprocess.check_call(
+                    ["git", "cat-file", "-e", commit + "^{commit}"],
```

Comment:
what scenario is this guarding against in which git output is not already sufficient or superior?

Reply:
This check whether the given commit exists; the `-e` option makes it exit with 0 if it's a valid object, or non-zero if not. Clearer, user-friendlier error messages (that also include a source name) are the goal here.

Reply:
isn't it better to report the error better when it happens rather than introducing a double-check antipattern?  by simply raising a "failed to checkout source X"?

also, this could be raised inside the except clause, which would carry additional context.